### PR TITLE
Bugfix/product deletion #5974

### LINKED ISSUE #5974: A product is unnecessary deleted from cart for logged in user after "out of stock error"
## Current behavior
A product is unnecessary deleted from cart for logged in user after "out of stock error"

## Expected behavior
A product shouldn't be deleted after an error

## Steps to reproduce the issue
1. Log in
2. Add product to cart with stock 1.
3. Log out
4. Log in
5. Add the same product to cart. You'll get a stock error and product will be deleted from cart

## Version of Vue Storefront
1.12.3
- [x] Vue Storefront
- [ ] Vue Storefront Next

## Can you handle fixing this bug by yourself?

- [x] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer. (doesn't apply to Next)
Pick one option.

- [x] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: Chrome, latest version
- OS: MacOS Big Sur 11.2.3
- Node: 13.14.0
- Code Version: b635c0deb119520461d6c8402e92cafa21eeb68c

### PR DESCRIPTION
**Pull request by SergeyDarnopykh**
### Related Issues

closes #5974

### Short Description of the PR
Fix the issue of product deletion from cart on logout->login on "out of stock" error

### Pull Request Checklist

- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [x] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [x] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- I tested manually my code and it works well with both:
- [x] Default Theme
- [x] Capybara Theme
- [x] I have written test cases for my code

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

**Review by lukaszjedrasik — COMMENTED** [review]
Please resolve conflict and then we can merge your PR 🚀

**Review by lukaszjedrasik — APPROVED** [review]

File: core/modules/cart/store/actions/quantityActions.ts [hunk 1]
```diff
@@ -7,12 +7,14 @@ const quantityActions = {
   async restoreQuantity ({ dispatch }, { product }) {
     const currentCartItem = await dispatch('getItem', { product })
     if (currentCartItem) {
-      Logger.log('Restoring qty after error' + product.sku + currentCartItem.prev_qty, 'cart')()
-      if (currentCartItem.prev_qty > 0) {
+      const prevQty = currentCartItem.prev_qty || product.prev_qty;
+
+      Logger.log('Restoring qty after error' + product.sku + ' ' + prevQty, 'cart', )()
```
Comment (Fifciu) [inline core/modules/cart/store/actions/quantityActions.ts:12]:
```suggestion
      Logger.log('Restoring qty after error ' + product.sku + ' ' + prevQty, 'cart')()
```

**Review by Fifciu — CHANGES_REQUESTED** [review]

File: core/modules/cart/store/actions/quantityActions.ts (hunk 1 above)
Comment (SergeyDarnopykh) [reply core/modules/cart/store/actions/quantityActions.ts]: ✅

**Review by Fifciu — APPROVED** [review]

**Comment by Fifciu** [comment]
Good job @SergeyDarnopykh, thanks!

**Pull request merged**